### ISSUE TITLE
Develop Harvard Dataverse DevOps Documentation

### ISSUE THREAD

**Issue by cmbz** [IQSS/dataverse.harvard.edu#235]
### Description
Develop comprehensive documentation of Harvard Dataverse-related DevOps processes, procedures, and resources.

**Goals**
- First draft complete by 2024/01/31
- TBD

Top level document location: https://docs.google.com/document/d/1kMQ8dkqLhjoJrOOYpawSq4CKpkSIXNNw2Cny7cmWk5U/edit?pli=1

**Comment by cmbz** [comment]
**2023/11/27**
- @landreev and @stevenwinship please give a sense of total size and then size for an upcoming sprint.

**Comment by landreev** [comment]
@cmbz Not sure how to estimate the total effort. With documentation it's usually as much as you're willing to spend. But let's schedule it for the next sprint and see how much we cover. Our "size33" label seems about right; more than a day's work, but definitely less than a full sprint/developer worth. 
I'm assuming it's going to be mostly my task *in the beginning*. I just need to go through the existing docs, mine and Kevin's and get them organized and accounted for, and/or refreshed if needed. The overall effort will depend on what we already have. I especially need to look more carefully into the docs that Kevin wrote and maintained, to make sure I'm not missing any. I've already been doing that between working on other stuff in the last couple of weeks. But would be great to be able to focus more on it.

@stevenwinship I'll be passing these docs to you as I go.

Generally, Kevin and I made a good effort to have all the useful knowledge about the prod. servers written down. But it is entirely possible that some things got out of sync. And I want to make sure I'm not giving Steven any information that's not accurate at this point. 

It was a good exercise earlier this month when I put together a quick doc of the more sensitive info relevant to the testing framework (keys, passwords, procedures). Having new people around helps with identifying things that are working fine for the current team, but not properly documented anywhere in one place. I'm looking forward to doing the same with the production docs.

**Comment by cmbz** [comment]
**2023/11/29**
- Added 33 sizing as per @landreev 's estimate: https://github.com/IQSS/dataverse.harvard.edu/issues/235#issuecomment-1831032624
- Moved into Sprint Ready

**Comment by landreev** [comment]
Great, I will take a closer look. 
And, as I promised, I'll put together a quick doc on the important scripts on the prod. boxes asap. 
Which I should also share with everybody on the team who has a prod. server login.

**Comment by landreev** [comment]
> ... And, as I promised, I'll put together a quick doc on the important scripts on the prod. boxes asap.

Started said doc. Shared with the `g.harvard.edu` accounts  of the prod. servers shell login owners. https://docs.google.com/document/d/16oimaDiQPmNNH14tVNww86M_-i8BDHmzavyrxlTgMMQ/edit

"started" being the key word.

**Comment by landreev** [comment]
Apologies for neglecting this ticket. I kept it in review because I kept thinking of more bits and pieces to add to the documentation. But there is really no need to keep this issue open forever. The parent document linked in the description is ok as is (I have only added one extra link to it). We should simply continue improving and extending the underlying individual documents. In particular I want to make an extra effort to document any remotely non-trivial scripts and solutions used to address devops issues that may be reused going forward. 
But, once again, there's no need to keep this issue opened, since its goal was to create this top-level, umbrella documentation, which it has achieved. So, I'm going to go ahead and close it.
